### PR TITLE
1.19.4/dev/nexoner

### PR DESCRIPTION
**Pull request by nexonerr**
# Description

Port to 1.19.4

## List of changes

- Port to 1.19.4
- Remove rain type setter inside `BiomeBuilder` (part of porting)

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code Refactoring (non-breaking change which improved the Code Quality)
- [ ] Breaking Code Refactoring (breaking change which improved the Code Quality)
- [ ] Documentation update (JavaDoc or Markdown change)

### CODE REVIEW

File: src/main/java/com/github/manasmods/manascore/api/tab/AbstractInventoryTab.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public void renderButton(PoseStack pPoseStack, int pMouseX, int pMouseY, float p
     protected abstract void renderIcon(PoseStack pPoseStack, int pMouseX, int pMouseY, float pPartialTick);
 
     @Override
-    protected void renderBg(PoseStack pPoseStack, Minecraft pMinecraft, int pMouseX, int pMouseY) {
+    public void renderBg(PoseStack pPoseStack, Minecraft pMinecraft, int pMouseX, int pMouseY) {
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/manascore/api/tab/AbstractInventoryTab.java:55]: Is there a reason why this should be public?

File: src/main/java/com/github/manasmods/manascore/tab/IInventoryTab.java [hunk 2]
```diff
@@ -23,4 +25,6 @@ default Predicate<Screen> isCurrentScreen() {
             return tab != null && tab.equals(this);
         };
     }
+
+    void renderBg(PoseStack pPoseStack, Minecraft pMinecraft, int pMouseX, int pMouseY);
```
Comment (Charismara) [inline src/main/java/com/github/manasmods/manascore/tab/IInventoryTab.java:29]: Why is this required?

**Review by Charismara — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/manasmods/manascore/tab/IInventoryTab.java (hunk 2 above)
Comment (nexonerr) [reply src/main/java/com/github/manasmods/manascore/tab/IInventoryTab.java]: I have no idea. Removed. IntelliJ might have automatically requested that since the renderBg method wasn't overriden anymore.

File: src/main/java/com/github/manasmods/manascore/api/tab/AbstractInventoryTab.java (hunk 1 above)
Comment (nexonerr) [reply src/main/java/com/github/manasmods/manascore/api/tab/AbstractInventoryTab.java]: Probably not, as the override seems to be gone, fixed in latest commit

**Comment by nexonerr** [comment]
> It looks good to me. Please fix the few things from Codacy and i'll merge it then

also done

**Review by Charismara — APPROVED** [review]

**Comment by Charismara** [comment]
Well done :)
